src/dsl/dsl.jl

```diff
@@ -72,7 +74,7 @@ function address_from_expression(lhs)
 end
 
 function desugar_tildes(expr)
-    trace_ref = GlobalRef(@__MODULE__, Symbol("@trace"))
+    trace_ref = Symbol("@trace")
```

Comment:
@georgematheos I believe @ztangent made this change recently to fix a problem where

1. you need to recognize `Gen.@trace`, not just `@trace`, and if someone did `const X = Gen`, you also need to recognize `X.@trace`

2. you don't want to recognize `@trace` when it's not in scope (i.e. if someone does `import Gen`, then `@trace` should not work -- they should need to do `Gen.@trace`).

Reply:
I realize this is *inserting* `@trace` into the code, but what if the user has `@trace` referring to something else?

Reply:
Yes, we need to make sure that this is a `GlobalRef` not just `@trace`.

Reply:
Okay, I'll switch this back.  I had thought it was slightly redundant since we wrap everything in `GlobalRef` in `resolve_or_expand_macros`/`resolve_gen_macros`, but I see what you mean, we need to be careful with namespaces.